pkg/api/targetgroup_deployment.go

```diff
@@ -0,0 +1,122 @@
+package api
+
+import (
+	"net/http"
+
+	"github.com/common-fate/apikit/apio"
+	"github.com/common-fate/common-fate/pkg/storage"
+	"github.com/common-fate/common-fate/pkg/targetgroup"
+	"github.com/common-fate/common-fate/pkg/types"
+)
+
+// Your GET endpoint
+// (GET /api/v1/target-group-deployments)
+func (a *API) ListTargetGroupDeployments(w http.ResponseWriter, r *http.Request) {
+
+	ctx := r.Context()
+	res := types.ListTargetGroupDeploymentsResponse{
+		Res: []types.TargetGroupDeployment{},
+	}
+
+	listTargetGroupDeployments := storage.ListTargetGroupDeployments{}
+
+	dbq, err := a.DB.Query(ctx, &listTargetGroupDeployments)
+	if err != nil {
```

Comment:
If this return ddb.ErrNoItems then the API will respond in 500

Instead check if the error is !ddb.ErrNoItems
```
	if err != nil && err != ddb.ErrNoItems {
		apio.Error(ctx, w, err)
		return
	}
```